### PR TITLE
ssrf: add makefile at ssrf level

### PR DESCRIPTION
**Pull request by janiversen**
add makefile at ssrf level to ease make calls
it can be called as:
make
make all
- will generate both desktop and mobile
make mobile
- will mobile
make
- will generate desktop

Signed-off-by: Jan Iversen <jani@apache.org>

### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [x ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

add subsurface/makefile to facilitate make calls
the makefile sets LIBRARY_PATH and call make in either mobile or desktop

this is a convenience function.

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+default_target: all
+
+.PHONY : default_target
+
+# Allow only one "make -f <foo>" at a time, but pass parallelism.
+.NOTPARALLEL:
+
+
+mobile:
+	cd build-mobile; LIBRARY_PATH=../install_root/lib make
```
Comment (dirkhh) [inline makefile]: this will fail until the first time build.sh was run and created that directory

File: makefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# Allow only one "make -f <foo>" at a time, but pass parallelism.
+.NOTPARALLEL:
+
+
+mobile:
+	cd build-mobile; LIBRARY_PATH=../install_root/lib make 
+	cd build-mobile; LIBRARY_PATH=../install_root/lib make install
+
+desktop:
+	cd build; LIBRARY_PATH=../install_root/lib make
```
Comment (dirkhh) [inline makefile]: ditto

**Review by dirkhh — COMMENTED** [review]
Hmm. I don't know.
Not sure which type of user / developer this would be useful for. The ones who can't be bothered to read even the most basic documentation, I guess?

File: makefile (hunk 2 above)
Comment (janiversen) [reply makefile]: It is just a convenience for others and me, who constantly compile the 2 versions.

But no big deal if you do not like, I can do the same with an alias.

File: makefile (hunk 1 above)
Comment (janiversen) [reply makefile]: correct there should be a test for that.

File: makefile (hunk 2 above)
Comment (dirkhh) [reply makefile]: It's not that I don't like it. It's that I wonder who'd use it.
But once it's implemented in a sane way so it actually works on a freshly checked out tree... I'm fine.
You don't necessarily have to call build.sh from here - just don't fail with a weird error if someone calls this without calling build.sh first

Comment (janiversen) [reply makefile]: Corrected

**Comment by dirkhh** [comment]
I don't think I'll use it, but I can see how someone might like it this way.

**Pull request merged**